Specs/Core/EllipsoidRhumbLineSpec.js

```diff
@@ -0,0 +1,581 @@
+        var end = new Cartographic(0.0, CesiumMath.PI / 2.4);
+
+        var rhumb = new EllipsoidRhumbLine(start, end, ellipsoid);
+
+        expect(0.0).not.toEqual(rhumb.heading);
+    });
+
+    it('heading works when going around the world at constant latitude', function() {
+        var ellipsoid = new Ellipsoid(6, 6, 6);
+        var start = new Cartographic(0.0, 0.3);
+        var end = new Cartographic(CesiumMath.PI_OVER_TWO, 0.3);
+
+        var rhumb = new EllipsoidRhumbLine(start, end, ellipsoid);
+
+        expect(CesiumMath.PI_OVER_TWO).toEqualEpsilon(rhumb.heading, CesiumMath.EPSILON12);
+
+        start = new Cartographic(3 * CesiumMath.PI_OVER_TWO, 0.3);
+        end = new Cartographic(CesiumMath.PI, 0.3);
+        var rhumb2 = new EllipsoidRhumbLine(start, end, ellipsoid);
+        expect(-CesiumMath.PI_OVER_TWO).toEqualEpsilon(rhumb2.heading, CesiumMath.EPSILON12);
+    });
+
+    it('computes heading for vertical lines', function() {
+        var ellipsoid = Ellipsoid.WGS84;
+        var start = new Cartographic(0.0, 1.2);
+        var end = new Cartographic(0.0, 1.5);
+
+        var rhumb = new EllipsoidRhumbLine(start, end, ellipsoid);
+        expect(0.0).toEqualEpsilon(rhumb.heading, CesiumMath.EPSILON12);
+
+        var rhumb2 = new EllipsoidRhumbLine(end, start, ellipsoid);
+        expect(CesiumMath.PI).toEqualEpsilon(rhumb2.heading, CesiumMath.EPSILON12);
+    });
+
+    it('computes no distance', function() {
+        var ellipsoid = new Ellipsoid(6, 6, 3);
+        var start = new Cartographic(CesiumMath.PI_OVER_TWO, 0.0);
+        var end = new Cartographic(CesiumMath.PI_OVER_TWO, 0.0);
+
+        var rhumb = new EllipsoidRhumbLine(start, end, ellipsoid);
+        expect(0).toEqualEpsilon(rhumb.surfaceDistance, CesiumMath.EPSILON12);
+        expect(0).toEqualEpsilon(rhumb.heading, CesiumMath.EPSILON12);
+    });
+
+    it('computes distance at equator', function() {
+        var ellipsoid = new Ellipsoid(6, 6, 3);
+        var start = new Cartographic(-CesiumMath.PI_OVER_FOUR, 0.0);
+        var end = new Cartographic(CesiumMath.PI_OVER_FOUR, 0.0);
+
+        var rhumb = new EllipsoidRhumbLine(start, end, ellipsoid);
+        expect(CesiumMath.PI_OVER_TWO * 6).toEqualEpsilon(rhumb.surfaceDistance, CesiumMath.EPSILON12);
+    });
+
+    it('computes distance at meridian', function() {
+        var ellipsoid = new Ellipsoid(6, 6, 6);
+        var fifteenDegrees = Math.PI / 12;
+        var start = new Cartographic(CesiumMath.PI_OVER_TWO, fifteenDegrees);
+        var fortyfiveDegrees = Math.PI / 4;
+        var end = new Cartographic(CesiumMath.PI_OVER_TWO, fortyfiveDegrees);
+
+        var rhumb = new EllipsoidRhumbLine(start, end, ellipsoid);
+        var thirtyDegrees = Math.PI / 6;
+        expect(thirtyDegrees * 6).toEqualEpsilon(rhumb.surfaceDistance, CesiumMath.EPSILON12);
+    });
+
+    it('computes distance for sphere 90 degrees along meridian and equator and check equality', function() {
```

Comment:
Maybe something like "computes equal distances for 90 degree arcs along meridians and equator"? I didn't connect the 90 degrees as the arc length until I read the whole spec.